scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5240 @@
+#    example: cdf of uniform betweeen 1.0 and np.nextafter(1.0, np.inf)
+#  check behavior of moment methods when moments are undefined/infinite -
+#    basically OK but needs tests
+#  investigate use of median
+#  implement symmetric distribution
+#  implement composite distribution
+#  implement wrapped distribution
+#  implement folded distribution
+#  implement double distribution
+#  profile/optimize
+#  general cleanup (choose keyword-only parameters)
+#  compare old/new distribution timing
+#  make video
+#  PR
+#  add array API support
+#  why does dist.ilogcdf(-100) not converge to bound? Check solver response to inf
+#  _chandrupatla_minimize should not report xm = fm = NaN when it fails
+#  integrate `logmoment` into `moment`? (Not hard, but enough time and code
+#   complexity to wait for reviewer feedback before adding.)
+#  Eliminate bracket_root error "`min <= a < b <= max` must be True"
+#  Test repr?
+#  use `median` information to improve integration? In some cases this will
+#   speed things up. If it's not needed, it may be about twice as slow. I think
+#   it should depend on the accuracy setting.
+#  in tests, check reference value against that produced using np.vectorize?
+#  add `axis` to `ks_1samp`
+#  Getting `default_rng` takes forever! OK to do it only when support is called?
+#  User tips for faster execution:
+#  - pass NumPy arrays
+#  - pass inputs of floating point type (not integers)
+#  - prefer NumPy scalars or 0d arrays over other size 1 arrays
+#  - pass no invalid parameters and disable invalid parameter checks with iv_profile
+#  - provide a Generator if you're going to do sampling
+#  add options for drawing parameters: log-spacing
+#  accuracy benchmark suite
+#  Should caches be attributes so we can more easily ensure that they are not
+#   modified when caching is turned off?
+#  Make ShiftedScaledDistribution more efficient - only process underlying
+#   distribution parameters as necessary.
+#  Reconsider `all_inclusive`
+#  Should process_parameters update kwargs rather than returning? Should we
+#   update parameters rather than setting to what process_parameters returns?
+
+# Questions:
+# 1.  I override `__getattr__` so that distribution parameters can be read as
+#     attributes. We don't want uses to try to change them.
+#     - To prevent replacements (dist.a = b), I could override `__setattr__`.
+#     - To prevent in-place modifications, `__getattr__` could return a copy,
+#       or it could set the WRITEABLE flag of the array to false.
+#     Which should I do?
+# 2.  `cache_policy` is supported in several methods where I imagine it being
+#     useful, but it needs to be tested. Before doing that:
+#     - What should the default value be?
+#     - What should the other values be?
+#     Or should we just eliminate this policy?
+# 3.  `validation_policy` is supported in a few places, but it should be checked for
+#     consistency. I have the same questions as for `cache_policy`.
```

Comment:
I think it's best if you leave it out for now, and focus on the "must-have" requirements. There policies might be nice to have, but at this point, it's probably a classic case of "premature optimization".

Reply:
Ok that makes sense. 

Perhaps something like adding a `__slots__` early on could be worth considering then as well, which is often very difficult to introduce later on.